scrapy/utils/reqser.py

```diff
@@ -40,11 +42,11 @@ def request_to_dict(request, spider=None):
     return d
 
 
-def request_from_dict(d, spider=None):
-    """Create Request object from a dict.
+def request_from_dict(d: dict, spider: Spider = None) -> Request:
```

Comment:
spider should be annotated as Optional. 

I also wonder about Request type annotation; subclasses may also be returned, but if someone expects a subclass to be returned by request_from_dict, mypy would be raising an error. I'm on fence here; it can be ok, as users can workaround it by using typing.cast. Thoughts?

Reply:
It is impossible to guarantee anything except `Request` here. If user expect subclass she should explicitly check it's type.